src/portfft/descriptor.hpp

```diff
@@ -491,18 +512,89 @@ class committed_descriptor {
     // Dummy parameter is needed as only partial specializations are allowed without specializing the containing class
     template <detail::level Lev, typename Dummy>
     struct inner {
-      static Scalar* execute(committed_descriptor& desc, dimension_struct& dimension_data);
+      static Scalar* execute(committed_descriptor& desc, std::vector<kernel_data_struct>& kernels);
     };
   };
 
   /**
    * Calculates twiddle factors for the implementation in use.
    *
-   * @param dimension_data data about the dimension for which twiddles are needed
+   * @param kernels vector of kernels
    * @return Scalar* USM pointer to the twiddle factors
    */
-  Scalar* calculate_twiddles(dimension_struct& dimension_data) {
-    return dispatch<calculate_twiddles_struct>(dimension_data.level, dimension_data);
+  Scalar* calculate_twiddles(detail::level level, std::vector<kernel_data_struct>& kernels) {
+    return dispatch<calculate_twiddles_struct>(level, kernels);
+  }
+
+  /**
+   * utility function to set specialization constants
+   * @tparam SubgroupSize Subgroup size
+   * @param top_level selected level of implementation
+   * @param prepared_vec vector as returned from prepare_implementation
+   * @param compute_direction direction of compute, as in forward or backward
+   * @param kernel_num which dimension are the kernels being built for
+   * @param is_compatible flag to be set if the kernels are compatible
+   * @return
+   */
+  template <Idx SubgroupSize>
+  std::vector<kernel_data_struct> set_specialization_constants(
+      detail::level top_level,
+      std::vector<std::tuple<detail::level, std::vector<sycl::kernel_id>, std::vector<Idx>>>& prepared_vec,
+      direction compute_direction, std::size_t kernel_num, bool& is_compatible) {
+    Scalar scale_factor = compute_direction == direction::FORWARD ? params.forward_scale : params.backward_scale;
+    std::size_t counter = 0;
+    bool take_conjugate_on_load = false;
+    bool take_conjugate_on_store = false;
```

Comment:
I think this would benefit from a minimal explanation of why `take_conjugate_on_*` are needed, and when they're applied in the FFT algorithm.

Reply:
This PR essentially changes the way our backward transforms work. 
Instead of expressing backward transform as $`IFFT=\sum x_i e^{\frac{2 * pi * i * j }{N}}`$ it calculates IFFT as - 
$`IFFT = conjugate(FFT(conjugate(input)))`$. Note how I have removed multiplying `-1` to the imaginary part of the twiddles on the basis of direction from everywhere, thus making the compute steps for forward and backward transforms  exactly the same. Our CT logic is now agnostic of direction. 

I would say this is slightly better than our current approach, and also lays ground work and makes it inline with how backward Bluestein transforms will work.

Reply:
I can add the above as comment  (maybe where those spec constants are declared), describing why those spec-constants are being used in the first place.

Reply:
I understand what it does, but it isn't clear for someone who doesn't have the benefit of seeing it in the context of this PR.

I think a short comment in the code along the lines of what you've said would be beneficial.

Reply:
This is Done

Reply:
Resolving comment as addressed